perceval/backends/_abstract_backends.py

```diff
@@ -70,6 +70,19 @@ def samples(self, count: int):
 
 
 class AProbAmpliBackend(ABackend):
+
+    def __init__(self):
+        super().__init__()
+        self._cache_iterator = dict()
```

Comment:
Could you add a `clear_iterator_cache()` method to free the memory if needed?

Reply:
Yes, I'm on it.